bindings/java/src/main/native/worker.cc

```diff
@@ -152,7 +152,44 @@ Java_org_openucx_jucx_ucp_UcpWorker_recvTaggedNonBlockingNative(JNIEnv *env, jcl
                                                 ucp_dt_make_contig(1), tag, tagMask,
                                                 recv_callback);
 
-    ucs_trace_req("JUCX: recv_nb request %p, msg size: %zu, tag: %ld", request, size, tag);
+    ucs_trace_req("JUCX: tag_recv_nb request %p, msg size: %zu, tag: %ld", request, size, tag);
+
+    return process_request(request, callback);
+}
+
+JNIEXPORT jobject JNICALL
+Java_org_openucx_jucx_ucp_UcpWorker_tagProbeNonBlockingNative(JNIEnv *env, jclass cls,
+                                                              jlong ucp_worker_ptr,
+                                                              jlong tag, jlong tag_mask,
+                                                              jboolean remove)
+{
+    ucp_tag_recv_info_t info_tag;
+    ucp_tag_message_h msg_tag = ucp_tag_probe_nb((ucp_worker_h)ucp_worker_ptr, tag, tag_mask,
+                                                 remove, &info_tag);
+    jobject result = NULL;
+
+    if (msg_tag != NULL) {
```

Comment:
also check for UCS_PTR_IS_ERR(_ptr)

Reply:
https://github.com/openucx/ucx/blob/master/src/ucp/api/ucp.h#L2650-L2652:

```
 @return NULL                      - No match found.
 * @return Message handle (not NULL) - If message is matched the message handle
 *                                     is returned.
```

Reply:
ok